internal/config/config_map_test.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+func TestFindEntry(t *testing.T) {
+	tests := []struct {
+		name    string
+		key     string
+		output  string
+		wantErr bool
+	}{
+		{
+			name:   "find key",
+			key:    "valid",
+			output: "present",
+		},
+		{
+			name:    "find key that is not present",
+			key:     "invalid",
+			wantErr: true,
+		},
+		{
+			name:   "find key with blank value",
+			key:    "blank",
+			output: "",
+		},
+		{
+			name:   "find key that has same content as a value",
+			key:    "same",
+			output: "logical",
```

Comment:
Without this change this would return `same` as after finding the first `same` which is the value for the `erroneous` key it would then select the next node which is the `same` key node.